### PR TITLE
Withdrawls - Change Amount type to GWei instead of Wei

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
Withdrawls - Change Amount type to GWei instead of Wei

## Fixed Issue(s)

#4920 
https://github.com/ethereum/execution-apis/pull/354

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.datatypes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+class GWeiTest {
+
```
Comment (jframe) [inline datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java:22]: Can some more test cases be added covering the from and to methods in GWei type

Comment (usmansaleem) [reply datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java:25]: Added some additional unit tests.

File: datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  @Test
+  void gWeiFromHexStringValue() {
+    final GWei oneGwei = GWei.fromHexString("0x0000000000000002");
+    assertThat(oneGwei.getValue()).isEqualTo(BigInteger.TWO);
+  }
+
+  @Test
+  void gWeiToHexString() {
+    final GWei oneGwei = GWei.of(UInt64.valueOf(2));
+    assertThat(oneGwei.toHexString()).isEqualTo("0x0000000000000002");
```
Comment (siladu) [inline datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java:60]: Does valueOf(0) return "0", "0x" or "0x0"?

**Review by siladu — APPROVED** [review]
LGTM (once you fix the failing tests)

File: datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java (hunk 2 above)
Comment (usmansaleem) [reply datatypes/src/test/java/org/hyperledger/besu/datatypes/GWeiTest.java:60]: I'll update the test, there is a `toShortHexString` method which returns "0x0".

**Review by jframe — APPROVED** [review]

**Pull request merged**